Anature/src/application/anatures/stats/Stats.java

```diff
@@ -253,34 +303,40 @@ private boolean levelGained()
 		int nextLevel = getLevel() + 1;
 		int requiredExperienceForLevel = requiredExperienceForLevel(nextLevel);
 
-		if(getExperiencePoints() >= requiredExperienceForLevel)
+		if(getTotalExperiencePoints() >= requiredExperienceForLevel)
 		{
-			return incrementLevel();
+			setLevel(nextLevel);
+			return levelStats();
 		}
+
 		return false;
 	}
 
-	private boolean incrementLevel()
+	private boolean levelStats()
 	{
-		mLevel = +1;
-
-		int hitPointsIncrese = (int) ((0.01 * getBaseHitPoints()) + (0.02 * (getIVHitPoints() + getEVHitPoints())));
-		addLevelHitPoints(hitPointsIncrese);
+		int hitPointsIncrease = (int) ((0.01 * (double) getBaseHitPoints()) + (0.02 * ((double) getIVHitPoints() + (double) getEVHitPoints())));
+		hitPointsIncrease = hitPointsIncrease == 0 ? 1 : hitPointsIncrease;
+		addLevelHitPoints(hitPointsIncrease);
```

Comment:
In addition to adding to the total hp, you need to increase the current hp to match. For example, say I had a level 15 Null with 76 current and total hp (aka 100%). It levels up, which increases the total hp value, but it didn't increase the current value. So when I go into battle, it no longer has 100% hp. When an anature levels up, its current hp is supposed to stay consistent with the previous percentage.

Reply:
Ok I added a heal method to the Stats class and added that line.

Reply:
That's not exactly what I meant. What I mean is, you have to increase the current hp with the same **percentage** as it was prior to leveling, not just add the increased hp points to the current hp. So if the Anature has 75 / 100 hp and levels up to have 200 total hp, then the current hp should be 150. With how the system is now, they would have 175.

Reply:
Almost fixed. So now the problem has to do with rounding. For example, I had a level 10 go up 9 levels. Before the leveling it had 75% hp but afterwards it had 66%. This is because the equation that handles the new current hp rounds down (not sure if it always goes down or only sometimes) instead of rounding up all the time. As such, the current hp would never increase.

Here is the runtime results I got using println:
Starting Percent (Lvl 10): 0.7435897435897436 Hp: 58 / 78
Percent to use: 0.7435897435897436 | (int) ( 79 * 0.7435897435897436 ) = 58
Percent to use: 0.7341772151898734 | (int) ( 80 * 0.7341772151898734 ) = 58
Percent to use: 0.725 | (int) ( 81 * 0.725 ) = 58
Percent to use: 0.7160493827160493 | (int) ( 82 * 0.7160493827160493 ) = 58
Percent to use: 0.7073170731707317 | (int) ( 83 * 0.7073170731707317 ) = 58
Percent to use: 0.6987951807228916 | (int) ( 84 * 0.6987951807228916 ) = 58
Percent to use: 0.6904761904761905 | (int) ( 85 * 0.6904761904761905 ) = 58
Percent to use: 0.6823529411764706 | (int) ( 86 * 0.6823529411764706 ) = 58
Percent to use: 0.6744186046511628 | (int) ( 87 * 0.6744186046511628 ) = 58
Ending Percent (Lvl 19): 0.6666666666666666 58 / 87

To fix this, change the value line to this:
int setHitPoitnsValue = (int) Math.ceil(currTotalHp * getHitPoitnsPercent);

This will always make it round upwards and while that does mean that things will be a bit off (using this code made an Anature with 74.4% hp go to 76.8% aka 64/86 to 73/95) its the best I could think of right now. If you can think of a way to guarantee it stays the right percent then go with that.